src/components/Personalization/createExecuteDecisions.js

```diff
@@ -49,21 +49,32 @@ const processMetas = (collect, logger, actionResults) => {
     finalMetas.push(meta);
   });
 
-  if (isNonEmptyArray(finalMetas)) {
-    // collect here can either be the function from createCollect or createViewCollect.
-    collect({ decisionsMeta: finalMetas });
-  }
+  return finalMetas;
 };
 
 export default ({ modules, logger, executeActions, collect }) => {
-  return decisions => {
+  return (decisions, returnDecisionsMeta = false) => {
     const actionResultsPromises = decisions.map(decision => {
       const actions = buildActions(decision);
 
       return executeActions(actions, modules, logger);
     });
     return Promise.all(actionResultsPromises)
-      .then(results => processMetas(collect, logger, results))
+      .then(results =>
+        processMetas(collect, logger, results, returnDecisionsMeta)
```

Comment:
`processMetas()` at line 27 doesn't seem to have the 4th parameter.

Reply:
@ninaceban Does this comment still apply?

Reply:
nope, this file was completely refactored. Thank you